src/components/ActionButtonGroup/ActionButtonGroup.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useMemo, isValidElement } from 'react';
+import { ExpandMore, ExpandLess } from '@mui/icons-material';
+
+import Button from '../Button';
+import Box from '../Box';
+
+export type ActionButtonGroupProps = {
+  /**
+   * The Hajimari buttons to include in the action buttons group.
+   */
+  children: React.ReactNode;
+}
+
+const ActionButtonGroup = ({ children }: ActionButtonGroupProps) => {
+  const [showingMore, setShowingMore] = useState(false);
+
+  const arrayChildren = React.Children.toArray(children);
+  let caretRenders = arrayChildren.length > 4;
+
+  const updateFirstButton = (child: React.ReactNode): React.ReactNode => {
+    const props = { fullWidth: true, variant: 'primary' }
+    return React.cloneElement(child as React.ReactElement, props);
+  }
+
+  const updateSecondaryButtons = (child: React.ReactNode): React.ReactNode => {
+    const props = { fullWidth: true, variant: 'filled' }
+    return React.cloneElement(child as React.ReactElement, props);
+  }
+
+  const variantCorrected = useMemo(() => {
+    return arrayChildren.map<React.ReactNode>((child, idx) => {
+      if (idx === 0 && isValidElement(child)) {
+        return updateFirstButton(child);
+      }
+      return updateSecondaryButtons(child);
+    });
+  }, [arrayChildren]);
+
+  const rowData = useMemo(() => {
```

Comment:
this was a big brain moment 😅  some explanatory comments might help the next eng?? edit: i guess reading the storybook comments before might have been helpful ooops

Reply:
(rowData => topButton/smallButtons)